unittests/test_utility.py

```diff
@@ -173,6 +173,11 @@ def test_samefile(self):
                                os.path.join(prefix, 'broken1'))
         shutil.rmtree(prefix)
 
+    def test_is_interactive(self):
+        # Set `sys.ps1` to immitate an interactive session
+        sys.ps1 = 'rfm>>> '
+        assert os_ext.is_interactive()
```

Comment:
You should restore `sys.ps1` to its original value or delete the attribute if it didn't exist. You might break other tests otherwise!

Reply:
To be complete you should also check the case where `sys.flags.interactive` is set to True. Again you should restore the values after you do your test.

Reply:
The `sys.flags.interactive` is readonly and cannot be adjusted. It's value is set based on the `-i` command line option when running `python`.